### PR TITLE
いろいろなメトリクスを付けて学習させる

### PR DESCRIPTION
**Pull request by halhorn**
- Metrics
    - accuracy
    - recall
    - precision
    - recall_topK: 予測の確率が大きい順にK個取ってきて、それを予測値として recall を計算
        - 最終的な製品で固定のK個をタグとしてサジェストする場合の指標
    - predicted_tag_count: 予測でしきい値を超えたものの数
- data
    - multi-hot の変換関数を rank3 以上のtensor にも対応させる
    - cache, prefetch で高速化

### CODE REVIEW

File: youtube8m/data.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 
 def multi_hot(indices):
-    return tf.reduce_sum(tf.one_hot(indices, CLASS_NUM), axis=0)
+    return tf.reduce_sum(tf.one_hot(indices, CLASS_NUM), axis=-2)
```
Comment (halhorn) [inline youtube8m/data.py]: indices の rank ごとに、 one-hot に変換したテンソルは以下の shape になります
rank1: [indice_count, class_num]
rank2: [batch_size, indice_count, class_num]

ここで indice_count の次元で reduce_sum したいので、該当する次元は0ではなく-2になります
